topydo/lib/ListFormat.py

```diff
@@ -74,3 +85,146 @@ def strip_placeholder_braces(p_matchobj):
     whitespace = p_matchobj.group('whitespace') or ''
 
     return before + '%' + placeholder + after + whitespace
+
+def str_clean(p_str):
+    clean_str = re.sub(r'\\%', '%', p_str)
+    clean_str = re.sub(' *\t *', '\t', clean_str)
+
+    return clean_str
+
+class ListFormatParser(object):
+    def __init__(self, p_todolist, p_format=None):
+        self.format_string = re.sub(r'\\t', '\t', p_format or config().list_format())
+        self.todolist = p_todolist
+        self.one_line = False
+        self.line_width = get_terminal_size().columns
+        self.placeholders = {
```

Comment:
Moving placeholders dict outside of the class will create unnecessary problem with `%i` and `%I` in my opinion, as they depend on `Todolist` object.